src/formatDate.js

```diff
@@ -8,7 +8,54 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator1 = fromFormat.splice(-1, 1);
+  const separator2 = toFormat.splice(-1, 1);
+
+  const changedDate = date.split(separator1[0]);
+  let year;
+  let month;
+  let day;
+  const result = [];
+
+  for (let data = 0; data < fromFormat.length; data++) {
+    if (fromFormat[data].includes('Y')) {
+      year = changedDate[data];
+    }
+
+    if (fromFormat[data].includes('M')) {
+      month = changedDate[data];
+    }
+
+    if (fromFormat[data].includes('D')) {
+      day = changedDate[data];
+    }
+  }
+
+  for (const d of toFormat) {
+    if (d.includes('Y')) {
+      let formattedYear;
+
+      if (d.length === 4 && year.length === 2) {
+        formattedYear = +year < 30 ? `20${year}` : `19${year}`;
```

Comment:
The logic for determining the century for two-digit years assumes a cutoff at 30. This may not be suitable for all use cases. Consider making this cutoff configurable or documenting this behavior clearly.